tools/export/uvision4.py

```diff
@@ -73,10 +73,14 @@ def generate(self, progen_build=False):
 
         # get flags from toolchain and apply
         project_data['tool_specific']['uvision']['misc'] = {}
-        # asm flags only, common are not valid within uvision project, they are armcc specific
-        project_data['tool_specific']['uvision']['misc']['asm_flags'] = list(set(self.progen_flags['asm_flags']))
+        # need to make this a string for progen. Only adds preprocessor when "macros" set
+        asm_flag_string = '--cpreproc --cpreproc_opts=-D__ASSERT_MSG,' + ",".join(
```

Comment:
Is this ``--cpreproc --cpreproc_opts`` needed? I recall progen does it automatically if macros are defined?